### PR TITLE
fix: modal logic for assignment reminder

### PR DESCRIPTION
**Pull request by sean-brydon**
Fixes: https://app.campsite.co/cal/posts/fua85mqwhijl

### CODE REVIEW

File: apps/web/modules/event-types/views/event-types-single-view.tsx [hunk 1]
```diff
@@ -420,6 +397,44 @@ const EventTypePage = (props: EventTypeSetupProps) => {
...
+      if (formMethods.getValues("assignAllTeamMembers") === true) {
+        return;
+      }
+
+      // Check if event is managed event type - skip if there is assigned users
+      const assignedUsers = formMethods.getValues("children");
+      const isManagedEventType = eventType.schedulingType === SchedulingType.MANAGED;
+      if (isManagedEventType && assignedUsers.length > 0) {
+        return;
+      }
```
Comment (sean-brydon) [inline apps/web/modules/event-types/views/event-types-single-view.tsx:414]: Use logic for assignees on managed event types

File: apps/web/modules/event-types/views/event-types-single-view.tsx [hunk 2]
```diff
@@ -420,6 +397,44 @@ const EventTypePage = (props: EventTypeSetupProps) => {
...
   } = formMethods;
 
+  useEffect(() => {
+    const handleRouteChange = (url: string) => {
+      const paths = url.split("/");
+
+      // We don't need to show the modal if the user is assigning all team members as they are already assigned
+      if (formMethods.getValues("assignAllTeamMembers") === true) {
+        return;
+      }
```
Comment (sean-brydon) [inline apps/web/modules/event-types/views/event-types-single-view.tsx:407]: If all team members toggle is checked we don't need to worry about showing the modal as there is users assigned but in a different way

File: apps/web/modules/event-types/views/event-types-single-view.tsx [hunk 3]
```diff
@@ -205,30 +206,6 @@ const EventTypePage = (props: EventTypeSetupProps) => {
 
   const router = useRouter();
 
-  useEffect(() => {
```
Comment (sean-brydon) [inline apps/web/modules/event-types/views/event-types-single-view.tsx:208]: Moved below so we have access to form values

**Review by joeauyeung — CHANGES_REQUESTED** [review]
Heads up if we enable the all member assignment but don't save the modal doesn't show up. I think this is going to cause a lot of confusion for users so I think it's a blocking issue.

**Comment by sean-brydon** [comment]
> Heads up if we enable the all member assignment but don't save the modal doesn't show up. I think this is going to cause a lot of confusion for users so I think it's a blocking issue.

I agree - me and @ciaranha were talking about this but i feel like this is a differnt modal we need to show when something hasnt been saved 🤔

**Comment by sean-brydon** [comment]
> > Heads up if we enable the all member assignment but don't save the modal doesn't show up. I think this is going to cause a lot of confusion for users so I think it's a blocking issue.
> 
> I agree - me and @ciaranha were talking about this but i feel like this is a differnt modal we need to show when something hasnt been saved 🤔

I'll fix this issue here and we can do the save modal later on

**Review by joeauyeung — APPROVED** [review]
LGTM

**Comment by joeauyeung** [comment]
> > Heads up if we enable the all member assignment but don't save the modal doesn't show up. I think this is going to cause a lot of confusion for users so I think it's a blocking issue.
> 
> I agree - me and @ciaranha were talking about this but i feel like this is a differnt modal we need to show when something hasnt been saved 🤔

That's true. It feels like we could look for any dirty fields in the form and show a modal with "Do you want to leave without saving your changes?"

**Review by hariombalhara — APPROVED** [review]
Tested. Works !!

**Pull request merged**